components/Tags.tsx

```diff
@@ -1,53 +1,97 @@
+import { isDefined } from "helpers/tinaHelpers";
 import { useEffect } from "react";
 import { resetPaginationPage } from "redux/slices/pagination";
 import { resetTags, setTags } from "redux/slices/selectedTags";
 import { useAppDispatch, useAppSelector } from "redux/typesHooks";
+import { useEditState } from "tinacms/dist/react";
+
+export interface Tag {
+  name: string;
+  postsCount: number;
+}
 
 interface Props {
-  uniqueTags: string[];
+  tags: Tag[];
+}
+
+interface TagItemProps {
+  isActive: boolean;
+  onClick: () => void;
+  tagName: string;
+  tagCount?: number;
 }
 
-const Tags = ({ uniqueTags }: Props) => {
+interface TagsListProps {
+  tags: TagItemProps[];
+}
+
+const MINIMUM_VISIBLE_COUNT = 2;
+const SINGLE_POST_COUNT = 1;
+
+const TagItem = ({
+  isActive,
+  onClick,
+  tagName,
+  tagCount,
+}: TagItemProps): JSX.Element => (
+  <li className={isActive ? "active" : ""} onClick={onClick}>
+    {tagCount ? `${tagName} [${tagCount}]` : tagName}
+  </li>
+);
+
+const TagsList = ({ tags }: TagsListProps): JSX.Element => (
+  <ul className="filter-tags-list">
+    {tags.map((tag) => (
+      <TagItem key={tag.tagName} {...tag} />
+    ))}
+  </ul>
+);
+
+const Tags = ({ tags }: Props) => {
   const selectedTags = useAppSelector((state) => state.selectedTags);
 
-  const dispatch = useAppDispatch();
+  const { edit } = useEditState();
 
-  const noneTagSelected = selectedTags.length === 0;
-  const allTagsSelected = uniqueTags.length === selectedTags.length;
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
     dispatch(resetPaginationPage());
   }, [dispatch, selectedTags]);
 
-  return (
-    <ul className="filter-tags-list">
-      <li
-        className={noneTagSelected || allTagsSelected ? "active" : ""}
-        onClick={() => dispatch(resetTags())}
-      >
-        ALL
-      </li>
-      {uniqueTags.map((uniqueTag) => (
-        <li
-          key={uniqueTag}
-          className={selectedTags.includes(uniqueTag) ? "active" : ""}
-          onClick={() => {
-            if (selectedTags.includes(uniqueTag)) {
-              const updatedTags = selectedTags.filter(
-                (tag) => tag !== uniqueTag
-              );
-
-              dispatch(setTags(updatedTags));
-            } else {
-              dispatch(setTags([...selectedTags, uniqueTag]));
-            }
-          }}
-        >
-          {uniqueTag}
-        </li>
-      ))}
-    </ul>
-  );
+  const tagSelect = (uniqueTag: string): void => {
```

Comment:
```suggestion
  const tagSelect = (uniqueTag: string) => {
```

Why do you add function returns type?
This is already inside the intellisense